ksql-engine/src/main/java/io/confluent/ksql/function/UdfCompiler.java

```diff
@@ -51,6 +53,22 @@
       .put(List.class, index -> typeConversionCode("List", index))
       .build();
 
+  // Templates used to generate the UDF code
+  private static final String genericTemplate =
+      "#TYPE arg#INDEX;\n"
+          + "if(args[#INDEX] == null) arg#INDEX = null;\n"
+          + "else if (args[#INDEX] instanceof #TYPE) arg#INDEX = (#TYPE)args[#INDEX];\n"
+          + "else if (args[#INDEX] instanceof String) arg#INDEX = "
+          + "#TYPE.valueOf((String)args[#INDEX]);\n";
```

Comment:
Is this line basically coercing an unmatched type into a string? Wouldn't it be better to just throw an exception?

Reply:
Yay - grokable code!

I'm with @apurvam though - implicit type coercion scares me.   Why do we need it? Why not just throw?

If we need this, then at the very least let's wrap this in a try/catch so we can throw a more meaningful error. Also, what if `#Type.valueOf(String)` does not exist?

Reply:
No - there are no unmatched types as the types need to be in `typeConverters` map.
For types that are not `Map` or `List` (as they are handled differently), it is converting it from a String to the type by using the `valueOf()` method, i.e., `Boolean.valueOf(String)`, `Long.valueOf(String)`